EntityFramework.Exceptions.Tests/SqlServerTests.cs

```diff
@@ -8,7 +8,12 @@ namespace EntityFramework.Exceptions.Tests
 {
     public class SqlServerTests : DatabaseTests, IClassFixture<SqlServerDemoContextFixture>, IDisposable
     {
-        public SqlServerTests(SqlServerDemoContextFixture fixture) : base(fixture.Context)
+        public SqlServerTests(SqlServerDemoContextFixture fixture) : base(fixture.ContextOptions)
+        {
+        }
+
+        [Fact(Skip = "Skipping as IDENTITY_INSERT must be set to ON to write IDs directly in db")]
```

Comment:
Without actually testing it how do you now that inserting two items with the same PK results in error 2601 or 2627 ? Did you test it manually?

Reply:
I'm not sure to understand your question, let me know if I failed to answer it. I suppose in this message that we're only talking about SqlServer.

Msdn tells us [that](https://docs.microsoft.com/en-us/sql/relational-databases/errors-events/database-engine-events-and-errors?view=sql-server-ver15#errors-2000-to-2999) about errors:
* 2601 `Cannot insert duplicate key row in object '%.*ls' with unique index '%.*ls'.`
* 2627 `Violation of %ls constraint '%.*ls'. Cannot insert duplicate key in object '%.*ls'.`

It's still a bit cryptic for me, so I went on another website:
* [2601](http://www.sql-server-helper.com/error-messages/msg-2601.aspx)
* [2627](http://www.sql-server-helper.com/error-messages/msg-2627.aspx)

From these two links, my understanding is that 2601 will trigger when we specify a unique constraint and that it's not respected. While 2627 will trigger for PK constraint not respected.

So I'd guess that inserting two items with the same PK always result in error 2627.
But I didn't manually test the exception for these two, I only launched the automated tests.

Reply:
Yes, that's what I wanted to know. I also think that 2627 is thrown when the PK is violated but it would be nice if this was tested at least manually. By the way what if you use `Context.Database.ExecuteSqlCommand` to set `IDENTITY_INSERT ` on and off before and after the test ?

Reply:
Sure, I did it in 37bad0544f that I just pushed, but it required a bit of work.
I put it in a separate commit so that if you feel that it's too complex or not needed, we can throw it away easily from this PR. But if you're okay with this commit, I'll squash it.

Reply:
Yes, it looks more complicated to me as I think it needs to be.